File: ClientApp/app/models/payroll/Shift.ts
import { Injectable, Pipe, PipeTransform } from '@angular/core';

import { PayrollCodeMap } from "./PayrollCodeMap";
import { CarerBooking } from './Booking';

export class Shift {
    public id: number;
    public carerCode: number;
    public contractCode: number;
    public sequence: number;
    public day: number;
    public start: Date;
    public finish: Date;
    public shiftMins: number;
    public unpaidMins: number;
    public validBreak: boolean;
    public breakPolicyId: number;

    public bookings: CarerBooking[];
    public availabilityType: PayrollCodeMap;
}

@Pipe({
    name: 'shiftOffsetFilter'
})
@Injectable()
export class ShiftOffsetFilter implements PipeTransform {
    public transform(shifts: Shift[], offset:number): Shift[] {
        return shifts.filter(shift => shift.day == offset);
    }
}

File: Data/Models/Shift.cs
namespace Blackwood.Access.Models
{
    using System.ComponentModel.DataAnnotations;
    using System;

    public class Shift
    {
        [Key]
        public int Id { get; set; }
        public int CarerCode { get; set; }
        public int? ContractCode { get; set; }
        public int Sequence { get; set; }
        public int Day { get; set; }
        public DateTime? Start { get; set; }
        public DateTime? Finish { get; set; }
        public int ShiftMins { get; set; }
        public int UnpaidMins { get; set; }
        public int BiggestGap { get; set; }
    }
}